### PR TITLE
Decouple csv reading from physical layer

### PR DESCRIPTION
**Pull request by monsterkrampe**

### CODE REVIEW

**Comment by monsterkrampe** [comment]
@ellmau you can take over from here.
Everything seems to work as intended. I'm not sure if all the lifetime madness is really necessary; maybe this can be simplified.
Also, the things from `io` should probably distributed to the physical and the logical layer. I think this should be possible with the current implementations.
I also did not check the performance of CSV reading.

**Comment by ellmau** [comment]
> As a follow up step, one could bring back the proper term parsing for the `Any` type and introduce an additional `String` type. But I think that this is out of scope of this PR.

I think so too, we should try to have smaller PRs anyway.

File: src/logical/execution/execution_engine.rs [hunk 1]
```diff
@@ -111,12 +111,19 @@ impl<Strategy: RuleSelectionStrategy> ExecutionEngine<Strategy> {
...
-                DataSource::DsvFile { file, delimiter } => TableSource::DSV {
-                    file: *file.clone(),
-                    delimiter: *delimiter,
-                },
+                DataSource::DsvFile { file, delimiter } => {
+                    let logical_types = analysis
+                        .predicate_types
+                        .get(predicate)
+                        .cloned()
+                        .unwrap_or_else(|| (0..arity).map(|_| LogicalTypeEnum::Any).collect());
```
Comment (mmarx) [inline src/logical/execution/execution_engine.rs:121]:
```rust
                    .unwrap_or_else(|| vec![LogicalTypeEnum::Any; *arity]),
```

**Review by mmarx — CHANGES_REQUESTED** [review]
Looks good, just a small simplification

**Review by mmarx — APPROVED** [review]

**Pull request merged**